File: in_Java/0729 My Calendar I.java
class MyCalendar {
    List<List<Integer>> base;
    public MyCalendar() {
        base = new ArrayList<>();
    }
    
    public boolean book(int start, int end) {
        List<Integer> cur = new ArrayList<>();
        cur.add(start); cur.add(end);
        if(base.size() == 0){
            base.add(cur);
            return true;
        }
        for(List<Integer> L : base){
            int a = L.get(0);
            int b = L.get(1);
            if(end <= a || start >= b) continue;
            else return false;
        }
        base.add(cur);
        return true;
    }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * MyCalendar obj = new MyCalendar();
 * boolean param_1 = obj.book(start,end);
 */

File: in_Python/0729 My Calendar I.py
class MyCalendar(object):

    def __init__(self):
        self.arr = []
    
    def overlap(self, a, b):
        return a[0] < b[1] and b[0] < a[1]
    def book(self, start, end):
        """
        :type start: int
        :type end: int
        :rtype: bool
        """
        cur = [start, end]
        if len(self.arr) == 0:
            self.arr.append(cur)
            return True
        else:
            for x in self.arr:
                if self.overlap(x, cur):
                    return False
            self.arr.append(cur)
            return True


# Your MyCalendar object will be instantiated and called as such:
# obj = MyCalendar()
# param_1 = obj.book(start,end)